### DISCUSSION TITLE
which language to write the deployment code(Javascript or Solidity)

### DISCUSSION THREAD

**Discussion by sakshammishra112** [Cyfrin/foundry-full-course-cu#1922] [Q&A]
Is it okay to write the deployment code in Solidity, rather than writing it in JavaScript? I want to know what is best nowadays because, in one of patrick's FreeCodeCamp video, he has used JavaScript to write the deployment code.

**Comment by urosognjenovic** [comment]
Foundry is a Solidity based framework so it uses Solidity for deployment scripts. At the moment, Foundry is the most popular framework.

Hardhat is a JavaScript based framework and it uses JS for deployment.

I guess learning both ways is the best for you as a smart contract developer because different teams use different frameworks.

**Comment by EngrPips** [comment]
It is always great if you are good all around, So I will say Learn as much as you can, manage your time and resources efficiently

**Comment by PatrickAlphaC** [comment]
It depends on your framework! As of recent, I do everything in foundry.
